src/Framework/Exception/InvalidDataProviderException.php

```diff
@@ -9,11 +9,39 @@
  */
 namespace PHPUnit\Framework;
 
+use Throwable;
+
 /**
  * @no-named-arguments Parameter names are not covered by the backward compatibility promise for PHPUnit
  *
  * @internal This class is not covered by the backward compatibility promise for PHPUnit
  */
 final class InvalidDataProviderException extends Exception
```

Comment:
in https://github.com/sebastianbergmann/phpunit/blob/a73ec4950712a4ff809ff730abe244f460160d11/src/Metadata/Api/DataProvider.php#L60-L64 there is a single use-case in which we use `InvalidDataProviderException` but do not reference a single dataprovider, therefore this method still has a `public __construct` inherited from the parent-class.

we could introduce another named-constructor for this use-case which allows us to turn the `InvalidDataProviderException::__construct()` private, in case you prefer.